qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -184,7 +184,7 @@ def result(self, timeout=None, wait=5):
         self._wait_for_submission()
         try:
             this_result = self._wait_for_job(timeout=timeout, wait=wait)
-        except TimeoutError as err:
+        except JobTimeoutError as err:
             # A timeout error retrieving the results does not imply the job
             # is failing. The job can be still running. This is why we are not
```

Comment:
We are [explaining](https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/backends/ibmq/ibmqjob.py#L104) that "we throw if an error prevents Qiskit from knowing the exact state of a Job". Here, we do the contrary. I vote for changing this behaviour to make it consistent and change the tests accordingly.